src/Disks/ObjectStorages/S3/S3ObjectStorage.cpp

```diff
@@ -243,8 +243,7 @@ void S3ObjectStorage::listObjects(const std::string & path, RelativePathsWithMet
 
     S3::ListObjectsV2Request request;
     request.SetBucket(uri.bucket);
-    if (path != "/")
-        request.SetPrefix(path);
+    request.SetPrefix(path);
```

Comment:
@antonio2368 I removed this condition (added in https://github.com/ClickHouse/ClickHouse/pull/67975), but looks like MinIO is happy anyway (you will find more context in the referenced issue)